tensorflow_datasets/testing/mocking.py

```diff
@@ -160,17 +178,16 @@ def _generate_random_array(self, feature, tensor_info):
     else:
       max_value = 255
 
+    dtype = tensor_info.dtype
     # Generate some random values, depending on the dtype
-    if tensor_info.dtype.is_integer:
-      return self._rgn.randint(0, max_value, shape)
-    elif tensor_info.dtype.is_floating:
-      return self._rgn.random_sample(shape)
-    elif tensor_info.dtype == tf.string:
+    if dtype.is_integer:
+      return self._rgn.randint(0, max_value, shape).astype(dtype.as_numpy_dtype)
```

Comment:
We need to properly cast the data here since `encode_example` would throw otherwise.